src/vs/workbench/contrib/scm/browser/scmViewPane.ts

```diff
@@ -2629,11 +2629,13 @@ export class SCMActionButton implements IDisposable {
 
 		if (button.secondaryCommands?.length) {
 			const actions: IAction[] = [];
-			for (const commandGroup of button.secondaryCommands) {
-				for (const command of commandGroup) {
+			for (let index = 0; index < button.secondaryCommands.length; index++) {
+				for (const command of button.secondaryCommands[index]) {
 					actions.push(new Action(command.id, command.title, undefined, true, async () => await executeButtonAction(command.id, ...(command.arguments || []))));
 				}
-				actions.push(new Separator());
+				if (index !== button.secondaryCommands.length - 1) {
+					actions.push(new Separator());
+				}
```

Comment:
Having encountered similar bugs myself many times and I am on the fence between a fix like here (using the index) and calling `actions.pop()` after the loop is done. Too much choice 🤷

Reply:
Thanks for the suggestion. Did not think of that. Will consider it.